apis/kubedb/v1alpha2/kafka_types.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright AppsCode Inc. and Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kmapi "kmodules.xyz/client-go/api/v1"
+	ofst "kmodules.xyz/offshoot-api/api/v1"
+)
+
+const (
+	ResourceCodeKafka     = "kf"
+	ResourceKindKafka     = "Kafka"
+	ResourceSingularKafka = "kafka"
+	ResourcePluralKafka   = "kafkas"
+)
+
+// Kafka is the Schema for the kafka API
+
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=kf,scope=Namespaced
+// +kubebuilder:printcolumn:name="Type",type="string",JSONPath=".apiVersion"
+// +kubebuilder:printcolumn:name="Version",type="string",JSONPath=".spec.version"
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.phase"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+type Kafka struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KafkaSpec   `json:"spec,omitempty"`
+	Status KafkaStatus `json:"status,omitempty"`
+}
+
+// KafkaSpec defines the desired state of Kafka
+type KafkaSpec struct {
+	// Version of Kafka to be deployed.
+	Version string `json:"version"`
+
+	// Number of instances to deploy for a Kafka database.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Kafka topology for node specification
+	// +optional
+	Topology *KafkaClusterTopology `json:"topology,omitempty"`
+
+	// StorageType can be durable (default) or ephemeral
+	StorageType StorageType `json:"storageType,omitempty"`
+
+	// Storage to specify how storage shall be used.
+	Storage *core.PersistentVolumeClaimSpec `json:"storage,omitempty"`
+
```

Comment:
Lets add EphemeralStorage too. 
`EphemeralStorage *core.EmptyDirVolumeSource `json:"ephemeralStorage,omitempty"``

Reply:
EphemeralStorage is handled with storage type. For now, It's not needed.